src/ophyd_async/core/signal.py

```diff
@@ -253,6 +257,52 @@ def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> No
     return _sim_backends[signal].set_callback(callback)
 
 
+async def verify_readable(
```

Comment:
I think this function should be made explicitly private with an _, I'm also not sure about the name, maybe `_verify_async_output`?

Reply:
This checks the output of `read` or `read_configuration` which is `Dict[str, Reading]` so how about `_verify_readings`?